core/modules/util/ConfigStore.h

```diff
@@ -0,0 +1,109 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/**
+ * @file
+ *
+ * @ingroup util
+ *
+ * @brief Provide common configuration management framework
+ *
+ * Manage czar and worker (xrdssi plugin) configuration files
+ *
+ * @author Fabrice Jammes, IN2P3/SLAC
+ */
+
+#ifndef LSST_QSERV_UTIL_CONFIGSTORE_H_
+#define LSST_QSERV_UTIL_CONFIGSTORE_H_
+
+// System headers
+#include <string>
+
+// Qserv headers
+#include "global/stringTypes.h"
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/** @brief Manage Qserv configuration
+ *
+ */
+class ConfigStore {
+public:
+
+    // TODO remove it
+    static ConfigStore& getInstance() {
+        static ConfigStore instance;
+        return instance;
+    }
+
+    /** Build a ConfigStore object from a configuration file
+     *
+     * @param configFilePath: path to Qserv configuration file
+     */
+    ConfigStore(std::string const& configFilePath) {
+        parseFile(configFilePath);
+    };
+
+    void parseFile(std::string const& configFilePath);
```

Comment:
Do you want to load configuration from more than one file? If not I'd make this method private. If you want to keep it private then explain it's behavior (how it behaves if the same parameter appears in more than one file).